metapool/prep.py

```diff
@@ -938,3 +907,178 @@ def demux_pre_prep(pre_prep):
+
+    # after all possible sample_files have been matched to sample_ids, ensure
+    # that the number of matching files for each sample_id is 2 and only 2,
+    # _assuming_ that one is an R1 file and the other is an R2 file.
+    msgs = []
+    for sample_id in results:
+        count = len(results[sample_id])
+        if count != 2:
+            msg = sorted(results[sample_id])
+            msgs.append(f"{sample_id} matched an unexpected number of samples "
+                        f"({msg})")
+    if msgs:
+        raise ValueError("\n".join(msgs))
+
+    return results, unmatched_sample_ids
+
+
+def _find_filtered_files(fp):
+    # assume that only fastq.gz files are desired and it's okay to ignore
+    # other odd files that we find.
+
+    # assume that all desired fastq.gz files will be stored in a directory
+    # named 'filtered_sequences'. This will automatically ignore files in
+    # 'only-adapter-filtered', 'zerofiles' and possibly other directories
+    # in the future as well.
+
+    # we expect the filepaths to be of the form:
+    # <project_name_w_qiita_id>/filtered_sequences/<fastq_file>
+    # However, we intentionally search across all directories to locate
+    # any fastq files w/in a filtered_sequences subdirectory, even if
+    # they don't match our expectation so that no potentially desirable
+    # file is silently filtered out.
+    files = glob(f"{fp}/**/filtered_sequences/*.fastq.gz", recursive=True)
+
+    by_project = defaultdict(list)
+
+    run_prefix_mapping = {}
+
+    count = 0
+
+    for fastq_fp in files:
+        # generate the full path to each file for reference.
+        full_path = abspath(fastq_fp)
+
+        # determine the run_prefix and orientation of the file.
+        # We do not want to silently ignore the file if its orientation is not
+        # R1 or R2.
+        run_prefix, orientation = get_run_prefix(basename(fastq_fp))
+        if orientation not in ['R1', 'R2']:
+            raise ValueError(f"{fastq_fp} does not appear to be a forward or "
+                             "reverse read file")
+
+        # process the relative path of each file to ensure each is of the
+        # the expected form. Report anything out of the ordinary so that no
+        # files are silently missed.
+
+        # remove fp from the beginning of the file's path. The remaining path
+        # should be of the form:
+        # <project_name_w_qiita_id>/filtered_sequences/<fastq_file>
+        tmp = fastq_fp.replace(fp, '')
+        # remove any leading and/or trailing '/' characters from the
+        # remaining path.
+        # use sep instead of '/' to be more platform independent.
+        tmp = tmp.strip(sep)
+        tmp = tmp.split(sep)
+
+        # tmp[1] != 'filtered_sequences' doesn't appear to be possible given
+        # the glob statement above, but we'll keep it here for safety.
+        if len(tmp) != 3 or tmp[1] != 'filtered_sequences':
```

Comment:
why 3? what would happen with amplicon?

Reply:
Since the samples for the amplicon runs are not demuxed, the fastq filenames for the runs are well-defined and predictable. Hence, there was no need to migrate the amplicon pathway to use this new code; it continues to use the path we broke out for it long ago.

Reply:
OK, just to confirm, this will also be OK for pipelines that do not produce index files (just R1-R2), right?

Reply:
Right, it will be okay for pipelines that produce no index files, just an I1 or even an I1 and I2. The host-filtering process should create just  R1 and R2 files in the filtered_sequences directory. Line 1001 above will cause any fastq file found that is an I1 or I2 or an orientation couldn't be determined to immediately raise an error. This will let us know when we've accidentally passed in a file path that contains unexpected files; this would usually be a sign that it's the wrong set of files for this code to be processing.